bwc-test/src/test/java/org/opensearch/customcodecs/bwc/CustomCodecsBwcCompatibilityIT.java

```diff
@@ -0,0 +1,262 @@
+        assertTrue(indexExists(index));
+        StringBuilder bulkRequestBody = new StringBuilder();
+        ObjectMapper objectMapper = new ObjectMapper();
+        int numberOfRequests = Randomness.get().nextInt(10);
+        while (numberOfRequests-- > 0) {
+            for (int i = 0; i < Randomness.get().nextInt(100); i++) {
+                Map<String, Map<String, String>> indexRequest = new HashMap<>();
+                indexRequest.put("index", new HashMap<>() {
+                    {
+                        put("_index", index);
+                    }
+                });
+                bulkRequestBody.append(objectMapper.writeValueAsString(indexRequest) + "\n");
+                bulkRequestBody.append(objectMapper.writeValueAsString(Song.randomSong().asJson()) + "\n");
+            }
+            List<Response> responses = RestHelper.requestAgainstAllNodes(
+                    testUserRestClient,
+                    "POST",
+                    "_bulk?refresh=wait_for",
+                    RestHelper.toHttpEntity(bulkRequestBody.toString())
+            );
+            responses.forEach(r -> assertEquals(200, r.getStatusLine().getStatusCode()));
+        }
+    }
+
+    /**
+     * Runs a matchAll query against the test index
+     *
+     * @param index index to search
+     */
+    private void searchMatchAll(String index) throws IOException {
+        String matchAllQuery = "{\n" + "    \"query\": {\n" + "        \"match_all\": {}\n" + "    }\n" + "}";
+        int numberOfRequests = Randomness.get().nextInt(10);
+        while (numberOfRequests-- > 0) {
+            List<Response> responses = RestHelper.requestAgainstAllNodes(
+                    testUserRestClient,
+                    "POST",
+                    index + "/_search",
+                    RestHelper.toHttpEntity(matchAllQuery)
+            );
+            responses.forEach(r -> assertEquals(200, r.getStatusLine().getStatusCode()));
+        }
+    }
+
+    /**
+     * Creates a test index if it does not exist already
+     *
+     * @param index index to create
+     */
+
+    private void createIndexIfNotExists(String index) throws IOException {
+
+        if (indexExists(index)) {
+            logger.info("Index {} already created for the domain", index);
+            return;
+        }
+
+        // creating index
+        createIndex(
+                index,
+                Settings.builder()
+                        .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                        .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                        .put("index.codec", randomFrom("zstd", "zstd_no_dict"))
+                        .put("index.codec.compression_level", randomIntBetween(1, 6))
+                        .build()
+        );
+        ensureGreen(index);
+    }
+
+    @AfterClass
+    public static void cleanUp() throws IOException {
+        OpenSearchRestTestCase.closeClients();
```

Comment:
No need to  call `OpenSearchRestTestCase.closeClients()`, it is annotated as `@AfterClass` and will be called automatically